### PR TITLE
Fix for #274

### PR DESCRIPTION
**Pull request by lablnet**
https://github.com/zestframework/Zest_Framework/issues/274

### CODE REVIEW

**Comment by lablnet** [comment]
@peter279k  I do not know why test failing.

**Comment by peter279k** [comment]
The Travis CI build is failed because the assertions are failed.

It seems that the regular expression patterns are not correct. And it should be checked.

The failed assertion messages are as follows:

```
1) Framework\Tests\StrTest::testHasLowerCase
Failed asserting that false is true.
/home/travis/build/zestframework/Zest_Framework/Tests/Data/StrTest.php:39
```

**Comment by lablnet** [comment]
> The Travis CI build is failed because the assertions are failed.
> 
> It seems that the regular expression patterns are not correct. And it should be checked.
> 
> The failed assertion messages are as follows:
> 
> ```
> 1) Framework\Tests\StrTest::testHasLowerCase
> Failed asserting that false is true.
> /home/travis/build/zestframework/Zest_Framework/Tests/Data/StrTest.php:39
> ```

Am sure "Regx" is correct, you may try to use online regx editor and manually check that in your Home.php" file it produce right result

i do not know what the hack problem with UnitTest

File: src/Data/Str.php [hunk 1]
```diff
@@ -133,8 +133,8 @@ public static function hasLowerCase(string $str)
         $all_words = preg_split('/\s+/', $str);
 
         foreach ($all_words as $word) {
-            if (!preg_match_all('~^\p{Lu}~u', $word)) {
-                // There is one upper
+            if (!preg_match_all('~\p{Ll}~u', $word)) {
```
Comment (peter279k) [inline src/Data/Str.php:136]: The problem is about the `preg_match_all` returns value.

The `preg_match_all` will return integer, I think we don't use the `!preg_match_all` to check whether the return value is.

It should use `preg_match_all(.....) === 0` condition.

Reference: https://www.php.net/manual/en/function.preg-match-all.php

**Comment by lablnet** [comment]
This `PR` is now closed due to better one #280

**Pull request closed without merging**